components/core/src/compressor_frontend/Buffer.hpp

```diff
@@ -0,0 +1,76 @@
+#ifndef COMPRESSOR_FRONTEND_BUFFER_HPP
+#define COMPRESSOR_FRONTEND_BUFFER_HPP
+
+// C++ libraries
+#include <cstdint>
+#include <vector>
+
+// Project Headers
+#include "Constants.hpp"
+
+/**
```

Comment:
* Class header comment should go above the class definition.
* Comment lines should be less than 80 characters.
* This description is vague. You should ideally explain why you would need a static and a dynamic buffer, grouped together in one class.